packages/graphql/tests/integration/filtering/relationship-to-abstract-types.int.test.ts

```diff
@@ -0,0 +1,1171 @@
+            readable: true,
+            charset: "alphabetic",
+        });
+        const movieRuntime = faker.number.int({ max: 100000 });
+        const movieScreenTime = faker.number.int({ max: 100000 });
+
+        const seriesTitle = generate({
+            readable: true,
+            charset: "alphabetic",
+        });
+        const seriesEpisodes = faker.number.int({ max: 100000 });
+        const seriesScreenTime = faker.number.int({ max: 100000 });
+
+        const query = `
+            query Actors($title: String) {
+                ${typeActor.plural}(where: { actedIn_NONE: { title: $title } }) {
+                    name
+                    actedIn {
+                        title
+                        ... on ${typeMovie} {
+                            runtime
+                        }
+                        ... on ${typeSeries} {
+                            episodes
+                        }
+                    }
+                }
+            }
+        `;
+
+        await session.run(
+            `
+                CREATE (a:${typeActor} { name: $actorName })
+                CREATE (m:${typeMovie} { title: $movieTitle, runtime:$movieRuntime })
+                CREATE (a)-[:ACTED_IN { screenTime: $movieScreenTime }]->(m)
+                CREATE (a)-[:ACTED_IN { screenTime: $seriesScreenTime }]->(:${typeSeries} { title: $seriesTitle, episodes: $seriesEpisodes })
+                CREATE (a2:${typeActor} { name: $actorName2 })
+                CREATE (a2)-[:ACTED_IN { screenTime: $movieScreenTime }]->(:${typeMovie} { title: $movieTitle2, runtime:$movieRuntime })
+                CREATE (a2)-[:ACTED_IN { screenTime: $movieScreenTime }]->(m)
+            `,
+            {
+                actorName,
+                actorName2,
+                movieTitle,
+                movieTitle2,
+                movieRuntime,
+                movieScreenTime,
+                seriesTitle,
+                seriesEpisodes,
+                seriesScreenTime,
+            }
+        );
+
+        const gqlResult = await graphql({
+            schema: await neoSchema.getSchema(),
+            source: query,
+            contextValue: neo4j.getContextValues(),
+            variableValues: { title: movieTitle2 },
+        });
+
+        expect(gqlResult.errors).toBeFalsy();
+
+        expect(gqlResult.data).toEqual({
+            [typeActor.plural]: [
+                {
+                    actedIn: expect.toIncludeSameMembers([
+                        {
+                            episodes: seriesEpisodes,
+                            title: seriesTitle,
+                        },
+                        {
+                            runtime: movieRuntime,
+                            title: movieTitle,
+                        },
+                    ]),
+                    name: actorName,
+                },
+            ],
+        });
+    });
+});
+
+describe("union relationships", () => {
```

Comment:
I think it could be easier to follow in separate files these test as the two `describe` are independent of each other.